src/components/cylc/graph/Graph.vue

```diff
@@ -0,0 +1,631 @@
+        'dagre',
+        'hierarchical'
+      ],
+      /**
+       * Cytoscape layout data. Controls things like the current layout, whether layout is frozen or not, etc.
+       */
+      layoutName: 'dagre',
+      layoutStopped: true,
+      layoutReady: false,
+      freeze: false,
+      /**
+       * Cytoscape tooltip plugin object.
+       */
+      tippy: null,
+      /**
+       * Graph data. These values are empty by default, then are populated later via watchers.
+       */
+      graphData: {
+        nodes: [],
+        edges: []
+      },
+      cytoscapeInstance: null
+    }
+  },
+
+  /**
+   * The watchers created by this component.
+   *
+   * The workflows property is mapped by Vuex (mapState). Whenever Vuex changes it, due to
+   * changes to the store, we are automatically notified. Even if the workflows object reference
+   * was not changed, but one of its properties was (deep: true).
+   *
+   * When the workflowUpdated method is executed, it will calculate the graphData, and change it
+   * once done.
+   *
+   * The graphData watcher will react to changes (not deep!) of graphData, and call the function
+   * updateGraph (debounced). This function is responsible for then - as per its name - update
+   * the graph, the layout, any other setting necessary once the nodes and edges have been added
+   * or removed.
+   */
+  watch: {
+    workflows: {
+      handler: function (newValue) {
+        this.workflowUpdated(newValue)
+      },
+      deep: true
+    },
+    graphData: function () {
+      this.updateGraph()
+    }
+  },
+
+  /**
+   * When this component is created, we simply create debounced functions
+   */
+  created () {
+    this.updateGraph = debounce(this.updateGraph_, 100)
+    this.workflowUpdated = debounce(this.workflowUpdated_, 100)
+  },
+
+  mounted () {
+    // create a div for cytoscape, this is an approach similar to vue-cytoscape
+    // but the vue-cytoscape component offers more than what we need, so we imitate
+    // part of its behaviour here, and ignore the rest and handle the graph and cytoscape
+    // settings ourselves.
+    const element = document.createElement('div')
+    element.setAttribute('class', 'cytoscape')
+    this.$el.appendChild(element)
+    this.cytoscapeInstance = Object.freeze(cytoscape(({
+      container: element
+    })))
+    this.registerExtensions()
+    cytoscape.use(dagre)
+    this.setupEventListeners()
+    this.runLayout(this.cytoscapeInstance)
+    this.setupUndoRedo(this.cytoscapeInstance)
+    this.setupPanzoom(this.cytoscapeInstance)
+    this.setupNavigator(this.cytoscapeInstance)
+    this.setupInteractivity(this.cytoscapeInstance)
+    this.setupHtmlLabel(this.cytoscapeInstance, states)
+    // set the initial zoom-level and pan (position) of the graph
+    // to set only the zoom level, call: this.cytoscapeInstance.zoom(0.5)
+    this.cytoscapeInstance.viewport({
+      zoom: 0.5,
+      pan: {
+        x: 100,
+        y: 100
+      }
+    })
```

Comment:
Also spent a long time searching for this one. Reaaaally bothered by having to always zoom out when opening the graph for my two simple `five` and `families2` workflows :smile:

Reply:
That's a bit nicer than the (100, 100) hack.